server/src/main/java/org/elasticsearch/index/fielddata/SortingNumericDocValues.java

```diff
@@ -70,15 +70,20 @@ protected final void resize(int newSize) {
         // copying.
         long oldValuesSizeInBytes = values.length * Long.BYTES;
         int newValuesLength = ArrayUtil.oversize(newSize, Long.BYTES);
```

Comment:
Actually the expression above: 

```
long oldValuesSizeInBytes = values.length * Long.BYTES;

```
Can overflow as well?

Reply:
Because `length` and `Long.BYTES` are both `int`, right?

Reply:
yes

Reply:
booooo. And you can't test that one easily.

Reply:
Yeah. I guess so. I think because this thing is designed for extension maybe we should do something else? These two new methods are only for testing which is fine if you are compositing this thing. But when you extend it its hard to know which to extend.

Reply:
Maybe we should be just add asserts in the code. I think this is what they are design for?